src/metis/http/gist.lua

```diff
@@ -0,0 +1,285 @@
+--- gist.lua - Gist client for ComputerCraft
+-- Made by JackMacWindows for CraftOS-PC and CC: Tweaked
+--
+-- @module cc.http.gist
+
+local expect = require and require("cc.expect").expect or dofile("/rom/modules/main/cc/expect.lua").expect
+
+if not http then
+    if _G.config ~= nil then error("Gist requires http API\nSet http_enable to true in the CraftOS-PC configuration")
+    else error("Gist requires http API\nSet http_enable to true in ComputerCraft's configuration") end
+end
+
+local gist = {}
+
+local function emptyfn() end -- to reduce memory/speed footprint when using empty functions
+
+-- Internal functions
+
+local function getGistFile(data)
+    if not data.truncated then return data.content else
+        local handle = http.get(data.raw_url)
+        if not handle then error("Could not connect to api.github.com.") end
+        if handle.getResponseCode() ~= 200 then
+            handle.close()
+            error("Failed to download file data.")
+        end
+        local d = handle.readAll()
+        handle.close()
+        return d
+    end
+end
+
+local function setTextColor(c) if term.isColor() then term.setTextColor(c) elseif c == colors.white or c == colors.yellow then term.setTextColor(colors.white) else term.setTextColor(colors.lightGray) end end
```

Comment:
I'm pretty lenient on line lengths, but 207 is beyond any sense of reason. I think a cleaner thing to do would be something like:

```lua
local whatever = colours.yellow
if term.isColour() then
  whatever = colours.white
end
```

Then just do `term.setTextColour(whatever)`.